dipy/segment/clustering.py

```diff
@@ -68,12 +70,13 @@ def __getitem__(self, idx):
         """
         if isinstance(idx, int) or isinstance(idx, np.integer):
             return self.refdata[self.indices[idx]]
-        elif type(idx) is slice:
+        elif isinstance(idx, slice):
             return [self.refdata[i] for i in self.indices[idx]]
-        elif type(idx) is list:
+        elif isinstance(idx, list):
             return [self[i] for i in idx]
 
-        raise TypeError("Index must be a int or a slice! Not " + str(type(idx)))
+        raise TypeError(
```

Comment:
It is probably a matter of taste but I find that setting the message in a variable beforehand reads better (in any cases you'll end up with two lines). For example:
```python
msg = "Index must be a int or a slice! Not " + str(type(idx))
raise TypeError(msg)
```